sat/apiclient/state_checker.py

```diff
@@ -0,0 +1,56 @@
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Client for querying the State Checker service
+"""
+
+import logging
+
+from sat.apiclient.gateway import APIError, APIGatewayClient
+
+LOGGER = logging.getLogger(__name__)
+
+class StateCheckerClient(APIGatewayClient):
+    base_resource_path = 'statechecker/jobstat/'
+
+    def get_all(self):
+        """Get all results from State Checker.
+
+        Returns:
+            A list of dictionaries where each dictionary pertains to a
+            single job.
+
+        Raises:
+            APIError: if there is a failure querying the State Checker API
+                or getting the required information from the response.
+        """
+        err_prefix = 'Failed to get State Checker data'
+        try:
+            return self.get('all').json()['jobstat']
```

Comment:
I am getting some weird behavior running through this code. The state checker service seems to be returning `200` code responses but are actually error cases, and as a result `APIError` is not being raised here as we would expect.

```
>>> resp = self.get('all')
>>> resp
<Response [200]>
```
But in the actual JSON there is an error:
```
>>> resp.json()
{'errorcode': 4, 'errormessage': 'Problem processing connection request', 'timereported': '2022-09-19T18:34:24', 'jobstat': []}
```

It seems like this should really be addressed in the `jobstat` service (codes of 4XX or 5XX will automatically raise an error). Maybe you could work around it here by raising an `APIError` if the returned JSON contained an `errorcode` field, but I wouldn't say that's an optimal solution.

```
        try:
            response = self.get('all').json()
            error_code = response.get('errorcode')
            if error_code is not None and error_code != 0:
                raise APIError(f'{response.get("errormessage", "error in state checker API")}')
            return response['jobstat']
        except APIError as err:
            raise APIError(f'{err_prefix}: {err}')
```